filebeat/harvester/log.go

```diff
@@ -137,6 +147,8 @@ func (h *Harvester) Harvest() {
 				Bytes:         bytesRead,
 				Text:          &text,
 				Fileinfo:      &info,
+				JSONFields:    jsonFields,
+				JSONConfig:    h.Config.JSON,
```

Comment:
do not include JSON fields if json not configured?

move line 133 for final decoding inside this if.